spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/core/mapping/SpannerPersistentEntityImplTests.java

```diff
@@ -214,32 +215,55 @@ public void doWithChildrenCollectionsTest() {
 		PropertyHandler<SpannerPersistentProperty> mockHandler = mock(PropertyHandler.class);
 		SpannerPersistentEntity spannerPersistentEntity =
 				this.spannerMappingContext.getPersistentEntity(ParentInRelationship.class);
-		doAnswer((Answer) invocation -> {
+		doAnswer(invocation -> {
 			String colName = ((SpannerPersistentProperty) invocation.getArgument(0))
 					.getColumnName();
 			assertTrue(colName.equals("childrenA") || colName.equals("childrenB"));
 			return null;
 		}).when(mockHandler).doWithPersistentProperty(any());
 		spannerPersistentEntity.doWithInterleavedProperties(mockHandler);
+		verify(mockHandler, times(2)).doWithPersistentProperty(any());
+	}
+
+	@Test(expected = SpannerDataException.class)
+	public void testParentChildPkNamesMismatch() {
+		this.spannerMappingContext
+				.getPersistentEntity(ParentInRelationshipMismatchedKeyName.class);
 	}
 
 	private static class ParentInRelationship {
 		@PrimaryKey
 		String id;
 
+		@Interleaved
 		List<ChildAInRelationship> childrenA;
 
+		@Interleaved
 		List<ChildBInRelationship> childrenB;
 	}
 
 	private static class ChildAInRelationship {
 		@PrimaryKey
 		String id;
+
+		@PrimaryKey(keyOrder = 2)
+		String id2;
 	}
 
 	private static class ChildBInRelationship {
 		@PrimaryKey
 		String id;
+
+		@PrimaryKey(keyOrder = 2)
+		String id2;
+	}
+
+	private static class ParentInRelationshipMismatchedKeyName {
+		@PrimaryKey
+		String idNameDifferentThanKids;
```

Comment:
s/Kids/Children

Reply:
change made; would you mind re-approving